kernel/mmu.c

```diff
@@ -1174,14 +1175,16 @@ static inline void do_backing_store_page_out(uintptr_t location)
 #endif /* CONFIG_DEMAND_PAGING_TIMING_HISTOGRAM */
 }
 
-/* Current implementation relies on interrupt locking to any prevent page table
- * access, which falls over if other CPUs are active. Addressing this is not
- * as simple as using spinlocks as regular memory reads/writes constitute
- * "access" in this sense.
- *
- * Current needs for demand paging are on uniprocessor systems.
+#if defined(CONFIG_SMP) && defined(CONFIG_DEMAND_PAGING_ALLOW_IRQ)
+/*
+ * SMP support is embryonic at this point. Some resources such as the
+ * scratch page would need to be per CPU, backing store driver execution
+ * would have to be confined to the faulting CPU, statistics would need
+ * to cope with access concurrency, etc. For now let's simply enforce global
+ * serialization across all CPUs with a mutex.
  */
-BUILD_ASSERT(!IS_ENABLED(CONFIG_SMP));
+static struct k_mutex z_mm_paging_lock;
```

Comment:
z_mm_paging_lock appears to be relying upon the BSS properties to initialize it. Please using either k_mutex_init(), or define it using K_MUTEX_DEFINE().